test/spawn.jl

```diff
@@ -173,14 +173,17 @@ exename = joinpath(JULIA_HOME, Base.julia_exename())
 if valgrind_off
     # If --trace-children=yes is passed to valgrind, we will get a
     # valgrind banner here, not "Hello World\n".
-    @test readall(pipe(`$exename -f -e 'println(STDERR,"Hello World")'`, stderr=`cat`)) == "Hello World\n"
+    @test readall(pipeline(`$exename -f -e 'println(STDERR,"Hello World")'`, stderr=`cat`)) == "Hello World\n"
+    p = Pipe()
+    run(pipeline(`$exename -f -e 'println(STDERR,"Hello World")'`, stderr = p))
```

Comment:
use `spawn` not `run` here to avoid forcing the kernel to buffer the output (the windows kernel is forcing the process to stay alive until there is a reader for the stderr pipe causing the hang on appveyor).

will push a fix shortly.